File: src/api/whois.ts
import axios from 'axios'
import { ElMessage } from 'element-plus'

// API 基础配置
const api = axios.create({
  baseURL: process.env.VUE_APP_API_BASE_URL || '/api',
  timeout: 15000,
  headers: {
    'Content-Type': 'application/json'
  }
})

// 响应拦截器
api.interceptors.response.use(
  response => response.data,
  error => {
    ElMessage.error(error.response?.data?.message || '服务器错误，请稍后重试')
    return Promise.reject(error)
  }
)

// 联系人信息接口
interface Contact {
  name?: string
  organization?: string
  email?: string
  phone?: string
  country?: string
  province?: string
  city?: string
}

// Whois 查询结果接口
export interface WhoisResult {
  available: boolean
  domain: string
  registrar?: string
  creationDate?: string
  expiryDate?: string
  status?: string[]
  nameServers?: string[]
  updatedDate?: string
  registrant?: Contact
  admin?: Contact
  tech?: Contact
  whoisServer?: string
  domainAge?: number
  contactEmail?: string
}

// Whois 查询接口
export const whoisQuery = async (domain: string): Promise<WhoisResult> => {
  try {
    return await api.get(`/whois/${domain}`)
  } catch (error) {
    console.error('Whois query failed:', error)
    throw error
  }
} 

File: server/types/whois.go
/*
 * @Author: AsisYu 2773943729@qq.com
 * @Date: 2025-01-18 22:34:01
 * @LastEditors: AsisYu 2773943729@qq.com
 * @LastEditTime: 2025-01-18 22:34:07
 * @FilePath: \dmainwhoseek\server\types\whois.go
 * @Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
 */
package types

// WhoisResponse 统一的WHOIS响应结构
type WhoisResponse struct {
	Available    bool     `json:"available"`
	Domain       string   `json:"domain"`
	Registrar    string   `json:"registrar"`
	CreateDate   string   `json:"creationDate"`
	ExpiryDate   string   `json:"expiryDate"`
	Status       []string `json:"status"`
	NameServers  []string `json:"nameServers"`
	UpdateDate   string   `json:"updatedDate"`
	Registrant   *Contact `json:"registrant,omitempty"`
	Admin        *Contact `json:"admin,omitempty"`
	Tech         *Contact `json:"tech,omitempty"`
	WhoisServer  string   `json:"whoisServer,omitempty"`
	DomainAge    int      `json:"domainAge,omitempty"`
	ContactEmail string   `json:"contactEmail,omitempty"`
}

type Contact struct {
	Name         string `json:"name,omitempty"`
	Organization string `json:"organization,omitempty"`
	Email        string `json:"email,omitempty"`
	Phone        string `json:"phone,omitempty"`
	Country      string `json:"country,omitempty"`
	Province     string `json:"province,omitempty"`
	City         string `json:"city,omitempty"`
}

// WhoisProvider WHOIS服务提供者接口
type WhoisProvider interface {
	Query(domain string) (*WhoisResponse, error, bool)
	Name() string
}
